src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -87,6 +97,24 @@ export function attachToEventDelegator(eventDelegator: EventDelegator): void {
   });
 }
 
+function isSamePageWithHash(absoluteHref : string) : boolean {
+  const hashIndex = absoluteHref.indexOf('#');
+  // Excluding case hash="#"
+  return hashIndex > -1 && absoluteHref.length > hashIndex + 1 &&
+  location.href.replace(location.hash, '') === absoluteHref.substring(0, hashIndex);
+}
+
+function performScrollToElementOnTheSamePage(absoluteHref : string) {
+  const hashIndex = absoluteHref.indexOf('#');
+  var hash = absoluteHref.substring(hashIndex); 
+
+  const urlInBrowser = location.hash.length > 1 ? location.href.replace(location.hash, hash) : location.href + hash;
```

Comment:
Could you clarify why we build `urlInBrowser` here instead of just using `absoluteHref` directly? Isn't that already the exact URL we want to use for `history.pushState`?

Reply:
You are right! I completely missed to change this piece of code 😄  it previously was somewhere else where there was no absolutehref. Thanks!